Chat_App/Core/Custom UI/ViewModel/ChatViewModel.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  ChatViewModel.swift
+//  test
+//
+//  Created by Ana Dzamelashvili on 5/23/23.
+//
+
+import Foundation
+import UIKit
+
+protocol ChatViewModelDelegate: AnyObject {
+    func messagesLoaded()
+}
+
+class ChatViewModel {
+    weak var delegate: ChatViewModelDelegate?
+    private let coreDataManager = CoreDataManager()
+    private lazy var sentMessages = [MessageEntity]()
+  
+    
+    func filteredMessages(currentUser: Int) -> [MessageEntity] {
```

Comment:
see? If you move currentUser to viewModel you can even omit a function parameter right there and since your current user will be stored right there inside viewModel you'll be able to call this function without a parameter at all, but renaming would be a good idea as well. Something like `filteredMessagesForCurrentUser()` would be a decent fit, but you can come up with your own naming